pymc_marketing/mmm/tvp.py

```diff
@@ -85,27 +85,54 @@
 
 """
 
+from typing import Annotated
+
 import numpy as np
 import numpy.typing as npt
 import pandas as pd
 import pymc as pm
 import pytensor.tensor as pt
+from pydantic import BaseModel, Field, InstanceOf
 from pymc.distributions.shape_utils import Dims
 
 from pymc_marketing.constants import DAYS_IN_YEAR
 
 
+class HSGPKwargs(BaseModel):
+    m: int = Field(200, description="Number of basis functions")
+    L: (
+        Annotated[
+            float,
+            Field(
+                gt=0,
+                description="""
+                Extent of basis functions. Set this to reflect the expected range of in+out-of-sample data
+                (considering that time-indices are zero-centered).Default is `X_mid * 2` (identical to `c=2` in HSGP)
+                """,
+            ),
+        ]
+        | None
+    ) = None
+    eta_lam: float = Field(1, gt=0, description="Exponential prior for the variance")
+    ls_mu: float = Field(
+        5, gt=0, description="Mean of the inverse gamma prior for the lengthscale"
+    )
+    ls_sigma: float = Field(
+        5,
+        gt=0,
+        description="Standard deviation of the inverse gamma prior for the lengthscale",
+    )
+    cov_func: InstanceOf[pm.gp.cov.Covariance] | None = Field(
+        None, description="Gaussian process Covariance function"
+    )
+
+
 def time_varying_prior(
     name: str,
     X: pt.sharedvar.TensorSharedVariable,
     dims: Dims,
     X_mid: int | float | None = None,
-    m: int = 200,
-    L: int | float | None = None,
-    eta_lam: float = 1,
-    ls_mu: float = 5,
-    ls_sigma: float = 5,
-    cov_func: pm.gp.cov.Covariance | None = None,
+    hsgp_kwargs: dict | HSGPKwargs | None = None,
```

Comment:
What do you think about restricting this to HSGPKwargs here and doing all the conversions somewhere else?

Reply:
Very good point! Changed it in https://github.com/pymc-labs/pymc-marketing/pull/809/commits/39ad8ce5d545fdefbac1f2eba2a9ca662cf67d18